packages/firestore/src/api/database.ts

```diff
@@ -379,6 +385,33 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
     }
   }
 
+  private mergeSettings(
+    settingsLiteral: firestore.Settings
+  ): firestore.Settings {
+    settingsLiteral.host =
+      settingsLiteral.host === undefined
+        ? this._settings.host
+        : settingsLiteral.host;
+    settingsLiteral.ssl =
+      settingsLiteral.ssl === undefined
+        ? this._settings.ssl
+        : settingsLiteral.ssl;
+    settingsLiteral.timestampsInSnapshots =
+      settingsLiteral.timestampsInSnapshots === undefined
+        ? this._settings.timestampsInSnapshots
+        : settingsLiteral.timestampsInSnapshots;
+    settingsLiteral.cacheSizeBytes =
+      settingsLiteral.cacheSizeBytes === undefined
+        ? this._settings.cacheSizeBytes
+        : settingsLiteral.cacheSizeBytes;
+    settingsLiteral.ignoreUndefinedProperties =
+      settingsLiteral.ignoreUndefinedProperties === undefined
+        ? this._settings.ignoreUndefinedProperties
+        : settingsLiteral.ignoreUndefinedProperties;
+
+    return settingsLiteral;
```

Comment:
return {...this._settings, ...settingsLiteral }

Reply:
done. Renamed `forceLongPolling` to `experimentalForceLongPolling` in order to use spread operator properly.